### PR TITLE
fix: multi-architecture container build

### PR DESCRIPTION
**Pull request by supermaxiste**
After a bit of research I found out the [eclipse-temurin for ARM](https://hub.docker.com/r/amd64/eclipse-temurin) is available, but unfortunately Dockerfiles do not allow conditional execution of their layers.
I split up the previous Dockerfile into two Dockerfiles, one for each architecture. The only different is the starting layer.
I also changed the GH Action to build and push twice (once per architecture).

### CODE REVIEW

**Review by cmdoret — COMMENTED** [review]
Instead of having 2 Dockerfiles, isn't it easier to do this?

```Dockerfile
ARG ARCH_BASE=eclipse-temurin:11-alpine
FROM $ARCH_BASE as jre-build
...
```

And then:

```sh
docker build . --build-arg ARCH_BASE=eclipse-temurin:11-alpine -t myshacl:latest
docker build . --build-arg ARCH_BASE=amd64/eclipse-temurin:11-alpine -t myshacl:amd64-latest
```

**Comment by supermaxiste** [comment]
Added your suggestion, definitely better now!
I had to leave the `ARCH_BASE` argument undefined in the image because we can't override values, but it's defined in both build steps in the Action. Concerning the tag, since we specify the platform in both build phases, both can be `latest`.

**Comment by rmfranken** [comment]
I would love to review this and am honored for the request, but honestly, I don't think I have the docker skills yet to review the "quality" of your code. I can test whether it runs, but beyond that I think Cyril is the main cheffe here :D

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -62,13 +62,28 @@ jobs:
...
-          build-args: VERSION=${{ steps.get_version.outputs.version_build }}
+          build-args: |
+            VERSION=${{ steps.get_version.outputs.version_build }}
+            ARCH_BASE=eclipse-temurin:11-alpine
+          outputs: type=image
+
+      - name: Build and push ARM Docker image
+        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671 #v4.0.0
+        with:
+          file: .docker/Dockerfile_ARM
```
Comment (cmdoret) [inline .github/workflows/docker-image.yml:81]:
```suggestion
          file: .docker/Dockerfile
```

Comment (supermaxiste) [reply .github/workflows/docker-image.yml]: The error you're getting is caused by `Docker` using the wrong context. To build the container you should point to the `Dockerfile` and set `.` as the context:

```
docker build -f .docker/Dockerfile -t shacl-test:latest --build-arg ARCH_BASE=eclipse-temurin:11-alpine --build-arg VERSION=v1.4.2 .
```

You also reminded me to update the `README` with the additional argument.

Comment (cmdoret) [reply .github/workflows/docker-image.yml]: Great, thank you! We should indeed have this command in the readme, it's quite tricky

**Review by cmdoret — APPROVED** [review]
LGTM, thanks for updating the docs!

**Pull request merged**